internal/controllers/clusterextension_controller.go

```diff
@@ -109,14 +113,11 @@ func (r *ClusterExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 
 	var existingExt = &ocv1alpha1.ClusterExtension{}
 	if err := r.Client.Get(ctx, req.NamespacedName, existingExt); err != nil {
-		return ctrl.Result{}, client.IgnoreNotFound(err)
+		return ctrl.Result{}, utilerrors.NewAggregate([]error{client.IgnoreNotFound(err), nil})
 	}
 
 	reconciledExt := existingExt.DeepCopy()
 	res, reconcileErr := r.reconcile(ctx, reconciledExt)
-	if reconcileErr != nil {
```

Comment:
It's better to exit early that do it after status changes. The below code assumes that reconcile was successful, so I'd revert this change.

Reply:
we can exit early but I'll need to look at making sure the right stuff is still set in the failure scenario -- was a cause of test failure.

Reply:
fixed in a53c47fe